operator/src/test/java/oracle/kubernetes/operator/MainTest.java

```diff
@@ -570,6 +576,141 @@ public void afterNamespaceAdded_scriptConfigMapIsDefined() {
     assertThat(getScriptMap(ns), notNullValue());
   }
 
+  @Test
+  public void withNamespaceList_onCreateStartNamespacesStep_nsWatchStartedEventCreatedWithExpectedMessage() {
+    defineSelectionStrategy(SelectionStrategy.List);
+    HelmAccessStub.defineVariable(HelmAccess.OPERATOR_DOMAIN_NAMESPACES,
+        String.join(",", NS_WEBLOGIC1, NS_WEBLOGIC2, NS_WEBLOGIC3));
+    testSupport.defineResources(NAMESPACE_WEBLOGIC1, NAMESPACE_WEBLOGIC2, NAMESPACE_WEBLOGIC3,
+        NAMESPACE_WEBLOGIC4, NAMESPACE_WEBLOGIC5);
+
+    List<String> namespaces = Arrays.asList(NS_WEBLOGIC1, NS_WEBLOGIC2, NS_WEBLOGIC3);
+    testSupport.runSteps(
+        createDomainRecheck().createStartNamespacesStep(namespaces));
+
+    for (String ns: namespaces) {
+      MatcherAssert.assertThat("Event NAMESPACE_WATCHING_STARTED message",
+          EventTestUtils.containsEventWithMessage(getEvents(testSupport),
```

Comment:
It's not clear to me what is happening here. What are you asserting? Why are you doing it in a loop?

For complex assertions, there are a number of strategies, including creating a custom matcher.

Reply:
it is to verify that the operator generates an event for each of the namespaces that are supposed to be started.